src/program.ts

```diff
@@ -3037,6 +3055,26 @@ export abstract class DeclaredElement extends Element {
         }
       }
     }
+    // TODO: field and property cannot override each other, have different codegen
+    // if (self.kind == ElementKind.FIELD && base.kind == ElementKind.PROPERTY_PROTOTYPE) {
+    //   // class A implement I, class B extends A implement I
+    //   let selfField = <Field>self;
+    //   let baseProperty = this.program.resolver.resolveProperty(<PropertyPrototype>base);
+    //   if (!selfField.internalGetterSignature || !selfField.internalSetterSignature || !baseProperty) {
+    //     return false;
+    //   }
+    //   let baseGetterInsance = baseProperty.getterInstance;
+    //   let baseSetterInsance = baseProperty.setterInstance;
+    //   if (baseGetterInsance && baseSetterInsance) {
+    //     if (!selfField.internalGetterSignature.isAssignableTo(baseGetterInsance.signature)
+    //       || !selfField.internalSetterSignature.isAssignableTo(baseSetterInsance.signature)) {
+    //       return false;
+    //     }
+    //   } else {
+    //     return false;
+    //   }
+    //   return true;
+    // }
```

Comment:
This is most likely not going to work because fields and properties have different code generation. In particular, getters and setters or properties are function calls, whereas a field is just load/store to a memory location. Getter and setter functions can be overloaded, yielding a virtual stub, while fields do not participate in overloading. Suggesting to revert what I commented out here, and update the tests accordingly so fields and properties are not mixed for now.

To give an outlook for the future, we'll likely want to remove the concept of fields and transparently make these properties with automatically generated getter and setter functions (a few bits into this direction already exist, but it's not yet finished), where these getters and setters are consequently used whenever the field is accessed. Conceptionally, fields then become properties with generated accessors (that can participate in virtual stubs) and a storage location behind. Hence we'll likely not need the `Field` kind / class anymore, and instead have `Property` with a flag `IS_FIELD` or so. Out of scope for this PR, though :)